torchtext/csrc/vectors.cpp

```diff
@@ -70,6 +87,231 @@ struct Vectors : torch::CustomClassHolder {
   int64_t __len__() { return stovec_.size(); }
 };
 
+// trim str from both ends (in place)
+static inline void _trim(std::string &s) {
+  // trim start
+  s.erase(s.begin(), std::find_if(s.begin(), s.end(),
+                                  [](int ch) { return !std::isspace(ch); }));
+  // trim end
+  s.erase(std::find_if(s.rbegin(), s.rend(),
+                       [](int ch) { return !std::isspace(ch); })
+              .base(),
+          s.end());
+}
+
+std::pair<int64_t, int64_t> _infer_shape(std::ifstream &fin,
+                                         const int64_t delimiter_ascii) {
+
+  int64_t num_lines = 0, vector_dim = -1;
+  std::vector<std::string> vec_str;
+  std::string line, word;
+
+  while (std::getline(fin, line)) {
+    vec_str.clear();
+    if (vector_dim == -1) {
+      _trim(line);
+      std::stringstream s(line);
+
+      // get rid of the token
+      std::getline(s, word, static_cast<char>(delimiter_ascii));
+
+      // we assume entries for vector are always seperated by ' '
+      while (std::getline(s, word, ' ')) {
+        vec_str.push_back(word);
+      }
+
+      // assuming word, [vector] format
+      if (vec_str.size() > 2) {
+        // the header present in some(w2v) formats contains two elements
+        vector_dim = vec_str.size();
+        num_lines++; // first element read
+      }
+    } else {
+      num_lines++;
+    }
+  }
+  fin.clear();
+  fin.seekg(0, std::ios::beg);
+
+  return std::make_pair(num_lines, vector_dim);
+}
+
+void _load_tokens_from_file_chunk(
+    const std::string &file_path, const int64_t start_line,
+    const int64_t num_lines, const int64_t vector_dim,
+    const int64_t delimiter_ascii,
+    std::promise<LoadedChunkVectorsTuple> &&promise) {
+  std::ifstream fin;
+  fin.open(file_path, std::ios::in);
+
+  std::vector<std::string> tokens;
+  std::vector<torch::Tensor> vectors;
+  std::vector<float> vec_float;
+  std::vector<std::string> dup_tokens;
+  std::unordered_set<std::string> tokens_set;
+  std::string line, token, vec_val;
+  int64_t num_vecs_loaded = 0;
+
+  tokens.reserve(num_lines);
+  vectors.reserve(num_lines);
+
+  // get to line we care about
+  for (int64_t i = 0; i < start_line; i++) {
+    std::getline(fin, line);
```

Comment:
You can probably just use http://www.cplusplus.com/reference/istream/istream/ignore/ instead of actually writing something into the line. Ignore characters one newline at a time.